design-docs/006-tag-delete.md

```diff
@@ -0,0 +1,91 @@
+# Deleting Tags
+
+We like our tags immutable. That gives Icechunk users a lot of power.
+For example, they can always cache anything coming out
+from an Icechunk session initiated on a tag:
+```python
+repo = icechunk.Repository.open(...)
+session = repo.readonly_session(tag="v42")
+store = session.store
+```
+
+All data read from `store` will be immutable, it can be cached for as
+long as needed. This is only true if tags cannot be edited. If
+somebody can edit `v42` and move it to a different snapshot,
```

Comment:
if we allow delete, then we effectively allow this workflow. We should warn somewhere that any caching should rely on snapshot ID, not tag name.